cmd/system-probe/modules/network_tracer.go

```diff
@@ -193,14 +194,17 @@ func (nt *networkTracer) Register(httpMux *module.Router) error {
 			maps = strings.Split(listMaps, ",")
 		}
 
-		ebpfMaps, err := nt.tracer.DebugEBPFMaps(maps...)
+		var ebpfMaps bytes.Buffer
+
+		err := nt.tracer.DebugEBPFMaps(&ebpfMaps, maps...)
+
 		if err != nil {
 			log.Errorf("unable to retrieve eBPF maps: %s", err)
 			w.WriteHeader(500)
 			return
 		}
 
-		utils.WriteAsJSON(w, ebpfMaps)
+		utils.WriteAsJSON(w, ebpfMaps.String())
```

Comment:
httpWriter itself implements writer, so you can pass it straight into DebugEbpfMaps. This will allow you to cut out the allocation in `var ebpfMaps bytes.Buffer`